### PR TITLE
fix(helm): Cleanup job after completed with ttlSecondsAfterFinished

### PR DESCRIPTION
**Pull request by aboutphilippe**
### SUMMARY

Following best practices from https://kubernetes.io/docs/concepts/workloads/controllers/job/#clean-up-finished-jobs-automatically
Added ttlSecondsAfterFinished to automatically cleanup job after completed

### TESTING INSTRUCTIONS
Deploy superset with helm
Upgrade helm chart, any change to init will not be reflected due to already existing job.
Now with ttlSecondsAfterFinished first job gets cleaned up

### CODE REVIEW

**Review by dpgaspar — COMMENTED** [review]
Thank you for contributing.

LGTM, but you need to bump the chart version

File: helm/superset/templates/init-job.yaml [hunk 1]
```diff
@@ -24,6 +24,7 @@ metadata:
     "helm.sh/hook-delete-policy": "before-hook-creation"
   namespace: {{ .Release.Namespace }}
 spec:
+  ttlSecondsAfterFinished: 0
```
Comment (craig-rueda) [inline helm/superset/templates/init-job.yaml:27]: Can you make this optional? We already have an annotation that controls the cleanup via Helm: 

```yaml
"helm.sh/hook-delete-policy": "before-hook-creation"
```

I've found it helpful to keep the completed job around after running in order to see logs, etc. What's worse, most log aggregators like DataDog won't scrape logs of a given container until it's been running for 30s, so if the thing goes away immediately, logs won't be viewable.

**Suggestion:**
```yaml
# vaules.yaml
init:
  ttlSecondsAfterFinished: ~
```


```
{{- if .Values.init.ttlSecondsAfterFinished }}
ttlSecondsAfterFinished: {{ .Values.init.ttlSecondsAfterFinished }}
{{- end }}
```

Comment (aboutphilippe) [reply helm/superset/templates/init-job.yaml]: Good call, that way in preview/dev environment we can still troubleshoot and production we can reduce pressure on api server.

**Review by craig-rueda — APPROVED** [review]

**Comment by craig-rueda** [comment]
Pls re-gen the chart's README with `helm-docs` (https://github.com/norwoodj/helm-docs#installation)

**Review by craig-rueda — APPROVED** [review]

**Comment by rusackas** [comment]
Here's the one-line diff from running `helm-docs`

```diff
diff --git a/helm/superset/README.md b/helm/superset/README.md
index 016bff992..9a6c334c3 100644
--- a/helm/superset/README.md
+++ b/helm/superset/README.md
@@ -23,7 +23,7 @@ NOTE: This file is generated by helm-docs: https://github.com/norwoodj/helm-docs

 # superset

-![Version: 0.9.2](https://img.shields.io/badge/Version-0.9.2-informational?style=flat-square)
+![Version: 0.9.3](https://img.shields.io/badge/Version-0.9.3-informational?style=flat-square)

 Apache Superset is a modern, enterprise-ready business intelligence web application

(END)
```

**Comment by rusackas** [comment]
@aboutphilippe wondering if you can apply that diff, or give @craig-rueda and/or I write access to your branch, to help get this through :)

**Comment by craig-rueda** [comment]
Closing for now

**Pull request closed without merging**